### PR TITLE
short-circuit GeneralPredicates

### PR DESCRIPTION
**Pull request by xiaoanyunfei**
**What type of PR is this?**
/kind cleanup
/kind design
/kind feature

**What this PR does / why we need it**:

There are several predicate functions in `GeneralPredicates`, and if we unset `AlwaysCheckAllPredicates`, all predicate functions in `GeneralPredicates` will be executed

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/scheduler/algorithm/predicates/predicates.go [hunk 1]
```diff
@@ -149,8 +149,17 @@ var (
 		CheckServiceAffinityPred, MaxEBSVolumeCountPred, MaxGCEPDVolumeCountPred, MaxCSIVolumeCountPred,
 		MaxAzureDiskVolumeCountPred, MaxCinderVolumeCountPred, CheckVolumeBindingPred, NoVolumeZoneConflictPred,
 		CheckNodeMemoryPressurePred, CheckNodePIDPressurePred, CheckNodeDiskPressurePred, MatchInterPodAffinityPred}
+	checkAllPredicates = false
```
Comment (draveness) [inline pkg/scheduler/algorithm/predicates/predicates.go]: Use global var does not sound like a good approach to me which could introduce unnecessary complexity, could you explain a bit more about the benefit?

Comment (xiaoanyunfei) [reply pkg/scheduler/algorithm/predicates/predicates.go]: Is any other way to short-circuit predicate functions in `GeneralPredicates` if not using global var?  I think restructuring `type FitPredicate func` is not a good choice.

**Review by liu-cong — COMMENTED** [review]
I see currently the `alwaysCheckAllPredicates` flag is checked by the factory and then only used in [generic_scheduler](https://github.com/kubernetes/kubernetes/blob/8523a790fe6df732e46b03488a5f3567e9d18b82/pkg/scheduler/core/generic_scheduler.go#L664).
However, with this change, this flag is going to be global. Just be mindful that this is going to be a behavior change.

The GeneralPredicates are called in multiple places. Is there a use case that certain callers want `alwaysCheckAllPredicates=true` while some other callers want `alwaysCheckAllPredicates=false`? If so, then this flag should not be global.

**Pull request closed without merging**